src/test/java/wolox/training/controller/BookControllerTest.java

```diff
@@ -0,0 +1,86 @@
+package wolox.training.controller;
+
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.utils.JsonUtil;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+public class BookControllerTest {
+
+	private static final String ISBN = "12356";
+	private static final Long ID_USER = 1l;
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	private BookRepository bookRepository;
+
+	private Book bookBuilder() {
+		Book bookBuilder = new Book();
+		bookBuilder.setId(ID_USER);
+		bookBuilder.setGenre("M");
+		bookBuilder.setImage("xxxx");
+		bookBuilder.setTitle("cien anios de soledad");
+		bookBuilder.setPublisher("NA");
+		bookBuilder.setYear("1990");
+		bookBuilder.setIsbn(ISBN);
+		bookBuilder.setPages(1);
+		bookBuilder.setSubtitle("NA");
+		return bookBuilder;
+	}
+
+	@Test
+	public void createBooks() throws Exception {
```

Comment:
You can use in the name of the tests the convention **given[condition]_when[Action]_then[Expected_reult]**, this must to be a description of the test case, example:

- givenBookWhenAllValidValuesThenStatus200